src/xdsl/mlir_converter.py

```diff
@@ -1,4 +1,5 @@
-import mlir.ir
+import xdsl.mlir as mlir
```

Comment:
I must admit, I don't fully understand how importing works exactly in python, but I am uncertain whether this will actually fix the problem. I am guessing that you would like to have a `import mlir_iterators as mlir` somewhere in the specific converter, which should overwrite the `mlir` that is used here to the one we actually want ot use. I did several such attempts trying to reimport a module using the same name and that never worked. And as mentioned, I don't actually understand in what sense this approach differs from mine other than having this extra file, which I don't quite understand the functionality off, since it does not seem to be adjustable to import from somewhere else.

Reply:
Haha, true, the most interesting part is missing! In Line 1, I just import `'mlir'`. What *is* interesting, is that this string can now be read from anywhere such as environment variables.

But I thought about this again. What I like about your approach is that it shows you don't *need* a string. I thought that maybe a combination of the two could be the best: keep a global variable, for example, named `mlir` in `src/xdsl/config.py`, which is initialized with `__import__('mlir', fromlist=['*'])`. Then, all xDSL code has to use that variable instead of the global module name. This is similar to your class member variable but not bound to a class instance. Users can then still overwrite which module to use by overwriting that variable.

On the other hand, using a string has the advantage that the default `mlir` package does not need to exist if the string is changed even before attempting to import it.